File: 2020/Day 01/1.hs
import Text.Printf
import Control.Arrow
import Data.Function
import Data.Functor

main :: IO ()
main = do
  fn <- solve <$> input
  fn 2 &printf "Silver: %d\n"
  fn 3 &printf "Gold: %d\n"

solve :: [Int] -> Int -> Int
solve xs n = snd.head.filter ((== 2020).fst)
  $mapM (const xs) [1..n] <&> (sum &&& product)

input :: IO [Int]
input = map read.lines
  <$> readFile "input.txt"

File: 2022/Day 02/1.py
Lines = [[{k: v - 3 if v > 3 else v 
    for v, k in enumerate("ABCXYZ", 1)}[x] for x in ln.split()] 
    for ln in open('2022/Day 02/input.txt')]

def Silver(f, g):
    return g + {
        f+1: 6, f-2: 6, f: 3,
    }.get(g, 0)


def Gold(f, g):
    return 3 * (g - 1) + {
        3: f+1 if f < 3 else 1,
        1: f-1 if f > 1 else 3,
    }.get(g, f)

for Fn in (Silver, Gold):
    print(f"{Fn.__name__}:", sum(Fn(*p) for p in Lines))